### PR TITLE
natron: 2.3.15 -> 2.4.0

### PR DESCRIPTION
**Pull request by vojta001**
I would also like to add myself as a maintainer of Natron, however my commit to become a maintainer currently waits for #115189. Either wait for it to be merged or I'll open another PR for it soon.

###### Motivation for this change
Update

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 121212` at ab06c1ba run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>natron</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/video/natron/default.nix:126:5:
```
    |
126 |     license = lib.licenses.gpl2;
    |     ^
```
</li>
  </ul>
</details>

File: pkgs/applications/video/natron/default.nix [hunk 1]
```diff
@@ -1,21 +1,21 @@
 { lib, stdenv, fetchurl, qt4, pkg-config, boost, expat, cairo, python2Packages,
   cmake, flex, bison, pango, librsvg, librevenge, libxml2, libcdr, libzip,
-  poppler, imagemagick, openexr, ffmpeg_3, opencolorio, openimageio,
+  poppler, imagemagick, openexr, ffmpeg_4, opencolorio, openimageio,
```
Comment (SuperSandro2000) [inline pkgs/applications/video/natron/default.nix:3]:
```suggestion
  poppler, imagemagick, openexr, ffmpeg, opencolorio, openimageio,
```

File: pkgs/applications/video/natron/default.nix [hunk 2]
```diff
@@ -55,32 +50,28 @@ let
...
       '';
     })
     ({
       pluginName = "io";
-      sha256 = "OQg6a5wNy9TFFySjmgd1subvXRxY/ZnSOCkaoUo+ZaA=";
+      sha256 = "sha256-tt+G2YqHQn8vPdiWtRYuOF3ffz9Qo19oD8Z9DsTDvQQ=";
       nativeBuildInputs = [ pkg-config ];
       buildInputs = [
-        libpng ffmpeg_3 openexr opencolorio openimageio boost libGL
+        libpng ffmpeg_4 openexr opencolorio openimageio boost libGL
```
Comment (SuperSandro2000) [inline pkgs/applications/video/natron/default.nix:68]:
```suggestion
        libpng ffmpeg openexr opencolorio openimageio boost libGL
```

**Review by SuperSandro2000 — COMMENTED** [review]
You can just cherry-pick the maintainer commit into this PR if you want.

**Comment by vojta001** [comment]
@SuperSandro2000 I changed the ffmpeg and cherry-picked the maintainers commit.

Reading @r-rmcgibbo's comment; do I understand [this file](https://github.com/NatronGitHub/Natron/blob/RB-2.4/LICENSE.txt) correctly that I should update Natron's license to `gpl2Plus`?

**Comment by SuperSandro2000** [comment]
Not quite. That file is the same for both variants but the code contains a license header that mentions or later which means it is plus.

**Comment by vojta001** [comment]
Ah, I understand. Fortunately, that seems to be the case
* [The last commit touching the license file](https://github.com/NatronGitHub/Natron/commit/9ab1bb99430a2430e5b051d6313676188e23d584)
* [LICENSE_SHORT.txt](https://github.com/NatronGitHub/Natron/blob/RB-2.4/LICENSE_SHORT.txt)

**Review by puffnfresh — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
@vojta001 please fix the build failure on arm64

**Comment by vojta001** [comment]
@SuperSandro2000 I am afraid upstream doesn't support ARM yet https://github.com/NatronGitHub/Natron/issues/277

File: pkgs/applications/video/natron/default.nix [hunk 3]
```diff
@@ -132,8 +123,8 @@ stdenv.mkDerivation {
       functionalities to Adobe After Effects and Nuke by The Foundry.
     '';
     homepage = "https://natron.fr/";
-    license = lib.licenses.gpl2;
-    maintainers = [ maintainers.puffnfresh ];
+    license = lib.licenses.gpl2Plus;
+    maintainers = with maintainers; [ puffnfresh vojta001 ];
     platforms = platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/applications/video/natron/default.nix:128]:
```suggestion
    platforms = platforms.linux;
    broken = stdenv.isAarch64;
```

**Comment by SuperSandro2000** [comment]
@vojta001 please resolve the merge conflict.

**Comment by vojta001** [comment]
Done, if you don't mind a merge commit. I can rebase instead

**Comment by vojta001** [comment]
Oh, but it does not build then, `pythonPackages.certifi` dropped support for Python 2

File: pkgs/applications/video/natron/default.nix [hunk 4]
```diff
@@ -1,21 +1,21 @@
 { lib, stdenv, fetchurl, qt4, pkg-config, boost, expat, cairo, python2Packages,
```
Comment (SuperSandro2000) [inline pkgs/applications/video/natron/default.nix:1]: Can we remove python2?

**Comment by SuperSandro2000** [comment]
@vojta001 please rebase to fix the merge conflict

**Comment by vojta001** [comment]
Neither Natron 2.3.15, nor Natron 2.4.0 can be built without Python 2, therefore I can't fix this PR.

There is some hope though – Natron 2.5 should support Python 3 NatronGithub/Natron#828

**Pull request closed without merging**